lib/decompress/zstd_decompress.c

```diff
@@ -433,60 +433,135 @@ static size_t ZSTD_decodeFrameHeader(ZSTD_DCtx* dctx, const void* src, size_t he
     return 0;
 }
 
+/**
+ * Contains the compressed frame size and an upper-bound for the decompressed frame size.
+ * Note: before using `compressedSize` you must check for errors using ZSTD_isError().
+ *       similarly, before using `decompressedBound`, you must check for errors using:
+ *          `decompressedBound` != ZSTD_CONTENTSIZE_UNKNOWN
+ */
+typedef struct {
+    size_t compressedSize;
+    unsigned long long decompressedBound;
+} ZSTD_frameSizeInfo;
```

Comment:
You'll save some code duplication, and make it harder miss setting a field if you use this helper function.

```c
static ZSTD_frameSizeInfo ZSTD_errorFrameSizeInfo(size_t err)
{
    ZSTD_frameSizeInfo frameSizeInfo;
    frameSizeInfo.compressedSize = err;
    frameSizeInfo.decompressedBound = ZSTD_CONTENTSIZE_UNKNOWN;
    return frameSizeInfo;
}
```

Reply:
Great idea! I'll add this.

Reply:
What do you think about this?

```
static ZSTD_frameSizeInfo ZSTD_errorFrameSizeInfo(size_t err)
{
    return ZSTD_createFrameSizeInfo(err, ZSTD_CONTENTSIZE_ERROR);
}

static ZSTD_frameSizeInfo ZSTD_createFrameSizeInfo(size_t compressedSize, unsigned long long decompressedBound) {
    ZSTD_frameSizeInfo frameSizeInfo;
    frameSizeInfo.compressedSize = compressedSize;
    frameSizeInfo.decompressedBound = decompressedBound;
    return frameSizeInfo;
}
```